### DISCUSSION TITLE
调用 client.writevalue() 写入信号点，导致内存增加很多

### DISCUSSION THREAD

**Discussion by 1198735966** [eclipse-milo/milo#1125] [General]
**Describe the bug**
A clear and concise description of behavior being reported and, if applicable, why it's incorrect.

**Expected behavior**
A clear and concise description of what you expected to happen instead.

**Logs and Packet Captures**
If applicable, add application logs or Wireshark captures to help explain your problem.

**Additional context**
Version of Eclipse Milo you are using, OPC UA client or software server you are interacting with, etc...

**Comment by kevinherron** [comment]
You're going to need to describe the issue in English and provide more details.

**Reply by 1198735966** [reply]
Calling client.writevalue() to write the signal point results in a significant increase in memory

**Reply by kevinherron** [reply]
You'll need to provide an example that demonstrates this.

Neither of the write examples demonstrate this and no other user has reported this.

**Reply by 1198735966** [reply]
public static void writeValue(UaClient client, String identifier, Object value) {
        try {
            LogUtil.writelog("=====写入【" + identifier + "】====开始", logName + "_" + identifier);
            //创建变量节点
            NodeId nodeId = new NodeId(2, identifier);
            //创建Variant对象和DataValue对象
            Variant v = new Variant(value);
            DataValue dataValue = new DataValue(v, null, null);
            StatusCode statusCode = client.writeValue(nodeId, dataValue).get();
            LogUtil.writelog("=====写入【" + identifier + "】====值：" + value + "【写入状态】：" + statusCode.isGood(), logName + "_" + identifier);
            Thread.sleep(500);
            DataValue valueR = client.readValue(0.0, TimestampsToReturn.Both, nodeId).get();
            LogUtil.writelog("=====写完读取【" + identifier + "】====值：" + valueR.getValue().getValue(), logName + "_" + identifier);
            if(valueR.getValue().getValue() != value){
                StatusCode statusCodeN = client.writeValue(nodeId, dataValue).get();
                LogUtil.writelog("=====重新写入【" + identifier + "】====值：" + value + "【写入状态】：" + statusCodeN.isGood(), logName + "_" + identifier);
                DataValue valueRR = client.readValue(0.0, TimestampsToReturn.Both, nodeId).get();
                LogUtil.writelog("=====重新写完读取【" + identifier + "】====值：" + valueRR.getValue().getValue(), logName + "_" + identifier);
            }
            LogUtil.writelog("=====写入【" + identifier + "】====结束", logName + "_" + identifier);

        } catch (Exception e) {
            LogUtil.writeExceptionLog(e, logName + "_" + identifier);
        }
    }

version 0.2.4

**Reply by 1198735966** [reply]
Must multithreading be used to call client read/write signal points？

**Reply by kevinherron** [reply]
Multithreading is not required.
